test/unit/domain/consents/{ID}/generateChallenge.test.ts

```diff
@@ -0,0 +1,214 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+import { Request } from '@hapi/hapi'
+import { consentDB } from '../../../../../src/lib/db'
+import { Consent } from '../../../../../src/model/consent'
+import { thirdPartyRequest } from '../../../../../src/lib/requests'
+// eslint-disable-next-line max-len
+import {
+  updateConsentCredential,
+  isConsentRequestInitiatedByValidSource,
+  putConsentId
+} from '../../../../../src/domain/consents/{ID}/generateChallenge'
+import { Enum } from '@mojaloop/central-services-shared'
+import Logger from '@mojaloop/central-services-logger'
+import { GenericRequestResponse } from '@mojaloop/sdk-standard-components'
+import { ExternalScope } from '../../../../../src/lib/scopes'
+
+// Declaring Mock Functions
+const mockPutConsents = jest.spyOn(thirdPartyRequest, 'putConsents')
+const mockConsentDbUpdate = jest.spyOn(consentDB, 'update')
+const mockLoggerPush = jest.spyOn(Logger, 'push')
+const mockLoggerError = jest.spyOn(Logger, 'error')
+
+/*
+ * Mock Request Resources
+ */
+// @ts-ignore
+const request: Request = {
+  headers: {
+    fspiopsource: 'pisp-2342-2233',
```

Comment:
Do we need to keep these all lower-case? Or can we do proper capitalization (e.g., `FSPIOPSource`) ?

Reply:
so in a different repo I found this being accessed as lowercase - but this is also just a test so either works

Reply:
Actually @jgeewax since we should be following the enum declaration the correct usage is 'fispiop-source' I think 
Have made a change to reflect that